source/includes/fundamentals/code-snippets/Monitoring.java

```diff
@@ -0,0 +1,169 @@
+        // Run a command to trigger a ClusterDescriptionChangedEvent event
+        collection.find().first();
+        // end monitor-cluster-example
+        mongoClient.close();
+    }
+
+    private void monitorConnectionPoolEvent() {
+        // start monitor-cp-example
+        ConnectionPoolLibrarian cpListener = new ConnectionPoolLibrarian();
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .applyToConnectionPoolSettings(builder ->
+                                builder.addConnectionPoolListener(cpListener))
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        MongoDatabase database = mongoClient.getDatabase(DATABASE);
+        MongoCollection<Document> collection = database.getCollection(COLLECTION);
+        // Run a command to trigger connection pool events
+        collection.find().first();
+        // end monitor-cp-example
+        mongoClient.close();
+    }
+
+}
+
+// start command-listener-impl
+class CommandTimer implements CommandListener {
+
+    private long startTime;
+    private long endTime;
+
+    @Override
+    public void commandStarted(final CommandStartedEvent event) {
+        this.startTime = System.nanoTime();
+    }
+
+    @Override
+    public void commandSucceeded(final CommandSucceededEvent event) {
+        TimeUnit currentTimeUnit = TimeUnit.MILLISECONDS;
+        this.endTime = System.nanoTime();
+        long duration = currentTimeUnit.convert(
+                (this.endTime - this.startTime),
+                TimeUnit.NANOSECONDS);
+        System.out.println(String.format("Command '%s' took %d %s",
+                event.getCommandName(),
+                duration,
+                currentTimeUnit.toString().toLowerCase()));
+    }
+
+    @Override
+    public void commandFailed(final CommandFailedEvent event) {
+        System.out.println(String.format("Failed execution of command '%s' with id %s",
+                event.getCommandName(),
+                event.getRequestId()));
+    }
+}
+// end command-listener-impl
+
+// start cluster-listener-impl
+class IsReadAndWrite implements ClusterListener {
+
+    private boolean isWritable;
+    private boolean isReadable;
+    
+    @Override
+    public void clusterDescriptionChanged(final ClusterDescriptionChangedEvent event) {
+        if (!isWritable) {
+            if (event.getNewDescription().hasWritableServer()) {
+                isWritable = true;
+                System.out.println("Able to write to server");
+            }
+        } else {
+            if (!event.getNewDescription().hasWritableServer()) {
+                isWritable = false;
+                System.out.println("Unable to write to server");
+            }
+        }
+
+        if (!isReadable) {
+            if (event.getNewDescription().hasReadableServer(ReadPreference.primary())) {
```

Comment:
Perhaps more interesting with `ReadPreference.secondary()`.  Otherwise, it's the same as `isWritable`.

Reply:
Thanks for the note. I agree this 'is readable' section of the class is redundant. I agree that this would be more interesting with `ReadPreference.secondary()` as was demonstrated in the previous Java docs, however by using `ReadPreference.secondary()` I fear that the example will be harder for readers to understand (not a straightforward explanation for what it is doing, 'writable primary and readable secondary' is less intuitive than 'can read' or 'can write'.

To reduce redundancy in the example, I have changed the example class to 'IsWritable'. This has the benefit of shortening the example code. 

Please let me know if you agree with this fix, or if you would prefer the 'ReadPreference.Secondary()` style example be included.

Fix: https://github.com/mongodb/docs-java/pull/88/commits/a21f0d21e7083ae2b4bbc0a4c8f526a20d2b8256